### PR TITLE
#94 UTF-8 as default charset for response body decoding

### PR DESCRIPTION
**Pull request by sanekyy**

### CODE REVIEW

File: lib/src/client/dart_http.dart [hunk 1]
```diff
@@ -10,12 +13,42 @@ class DartHttp implements HttpHandler {
...
+  /// Defaults to [UTF-8] if the headers don't specify a charset or if that
+  /// charset is unknown.
+  Encoding _encodingForHeaders(Map<String, String> headers) =>
+      encodingForCharset(_contentTypeForHeaders(headers).parameters['charset']);
+
+  /// Returns the [Encoding] that corresponds to [charset].
+  ///
+  /// Returns [fallback] if [charset] is null or if no [Encoding] was found that
+  /// corresponds to [charset].
+  Encoding encodingForCharset(String charset, [Encoding fallback = utf8]) {
```
Comment (f3ath) [inline lib/src/client/dart_http.dart]: Does it have to be public?

File: lib/src/client/dart_http.dart [hunk 2]
```diff
@@ -10,12 +13,42 @@ class DartHttp implements HttpHandler {
     final response = await _send(Request(request.method, request.uri)
       ..headers.addAll(request.headers)
       ..body = request.body);
-    return HttpResponse(response.statusCode,
-        body: response.body, headers: response.headers);
+    final responseBody =
+        _encodingForHeaders(response.headers).decode(response.bodyBytes);
```
Comment (f3ath) [inline lib/src/client/dart_http.dart:18]: Could you add a test for this change please?

**Review by f3ath — CHANGES_REQUESTED** [review]

File: lib/src/client/dart_http.dart (hunk 1 above)
Comment (sanekyy) [reply lib/src/client/dart_http.dart]: No. Fixed.

File: lib/src/client/dart_http.dart (hunk 2 above)
Comment (sanekyy) [reply lib/src/client/dart_http.dart:18]: Done

**Comment by f3ath** [comment]
Thank you!

**Pull request merged**